titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/common/reactor/MonoMethodBridge.java

```diff
@@ -33,15 +37,29 @@
 class MonoMethodBridge<GRPC_STUB extends AbstractStub<GRPC_STUB>> implements Function<Object[], Publisher> {
 
     private final Method grpcMethod;
+    private final int grpcArgPos;
+    private final int callMetadataPos;
+    private final CallMetadataResolver callMetadataResolver;
+    private final GRPC_STUB grpcStub;
     private final boolean emptyToVoidReply;
-    private final Supplier<GRPC_STUB> grpcStubFactory;
+    private final Duration timeout;
     private final Duration reactorTimeout;
 
-    MonoMethodBridge(Method reactMethod, Method grpcMethod, Supplier<GRPC_STUB> grpcStubFactory, Duration reactorTimeout) {
+    MonoMethodBridge(Method reactMethod,
+                     Method grpcMethod,
+                     int grpcArgPos,
+                     int callMetadataPos,
+                     CallMetadataResolver callMetadataResolver,
+                     GRPC_STUB grpcStub,
+                     Duration timeout) {
         this.grpcMethod = grpcMethod;
+        this.grpcArgPos = grpcArgPos;
```

Comment:
same comment from before about documenting that `argPos < 0` means `Empty`?

Reply:
Fixed